File: hackerrank/30 days of code/day 8, dictionary and maps/dictionary_and_maps.py
# https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem?isFullScreen=true

from sys import stdin


def main():
    n = int(input())
    phone_book = init_phone_book(n)
    query_names(phone_book)


def init_phone_book(n):
    phone_book = {}
    for _ in range(n):
        name, phone = input().split()
        phone_book[name] = phone
    return phone_book


def query_names(phone_book):
    for name in stdin:
        name = name.rstrip()
        if value := phone_book.get(name):
            print(f"{name}={value}")
        else:
            print('Not found')


if __name__ == '__main__':
    main()

File: hackerrank/30 days of code/day 8, dictionary and maps/dictionary_and_maps.ts
// https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem?isFullScreen=true

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString: string = '';
let inputLines: string[] = [];
let currentLine: number = 0;

process.stdin.on('data', function (inputStdin: string): void {
  inputString += inputStdin;
});

process.stdin.on('end', function (): void {
  inputLines = inputString.split('\n');
  inputString = '';
  main();
});

function readLine(): string {
  return inputLines[currentLine++];
}

//////////////////////////////////////////////////

function main() {
  let n: number = +readLine();
  let phoneBook: { [key: string]: string } = initPhoneBook(n);
  queryNames(phoneBook);
}

function initPhoneBook(n: number): { [key: string]: string } {
  let phoneBook: { [key: string]: string } = {};
  while (n-- > 0) {
    const [name, phone] = readLine().split(' ');
    phoneBook[name] = phone;
  }
  return phoneBook;
}

function queryNames(phoneBook: { [key: string]: string }) {
  for (let name; (name = readLine()); )
    if (phoneBook[name])
      console.log(`${name}=${phoneBook[name]}`);
    else
      console.log('Not found');
}
